src/common/hooks/use-make-stx-transfer.ts

```diff
@@ -0,0 +1,90 @@
+import { useRecoilCallback, useRecoilValue } from 'recoil';
+import { stacksNetworkStore } from '@store/recoil/networks';
+import { currentAccountStore } from '@store/recoil/wallet';
+import { correctNonceStore } from '@store/recoil/api';
+import { makeSTXTokenTransfer, StacksTransaction } from '@stacks/transactions';
+import BN from 'bn.js';
+import { stxToMicroStx } from '@stacks/ui-utils';
+import { useLoading } from '@common/hooks/use-loading';
+import { useEffect } from 'react';
+import { useMakeAssetTransfer } from '@common/hooks/use-asset-transfer';
+import { selectedAssetStore } from '@store/recoil/asset-search';
+
+interface TokenTransferParams {
+  amount: number;
+  recipient: string;
+}
+
+export function useMakeStxTransfer() {
+  return useRecoilCallback(({ snapshot }) => async (params: TokenTransferParams) => {
+    const { amount, recipient } = params;
+    const stacksNetwork = await snapshot.getPromise(stacksNetworkStore);
+    const account = await snapshot.getPromise(currentAccountStore);
+    const nonce = await snapshot.getPromise(correctNonceStore);
+
+    if (account) {
+      return await makeSTXTokenTransfer({
```

Comment:
unneeded await?

Reply:
It is an async method - it fetches the nonce and fee rate, or in this case just the fee rate.

Reply:
Yes, but it's returning to an async function